contracts/optimistic-ethereum/OVM/execution/OVM_SafetyChecker.sol

```diff
@@ -6,6 +6,14 @@ import { iOVM_SafetyChecker } from "../../iOVM/execution/iOVM_SafetyChecker.sol"
 
 /**
  * @title OVM_SafetyChecker
+ * @dev  The Safety Checker ensures that contracts deployed on L2 do not contain any
+ * "unsafe" opcodes. An opcode is considered unsafe if it would access state variables which
+ * are specific to the environment (ie. L1 or L2) in which it is executed, as this could be used
+ * to "escape the sandbox" of the OVM, resulting in non-deterministic fraud proofs. 
+ * That is, an attacker wouuld be able to "prove fraud" on an honestly applied transaction.
+ *
+ * This contract compiles to EVM bytecode.
```

Comment:
One potential sticking point is that technically, OVM code is a subset of EVM bytecode.  Further, we have cases where some L2 contracts use the L1 compiler, and others use our fork.  I lean towards being overly verbose and specifying all of this.

I also like having this be at the top of the docstrings.  If we wanted to avoid them getting too long, I guess it could be 1 line, something like 
```
* Compiler used: {EVM,OVM,BOTH}.  Deploy target: {EVM,OVM,BOTH}.
```

Reply:
I kind of addressed that OVM < EVM subset idea here: https://github.com/ethereum-optimism/contracts-v2/blob/42497aa6775c43b0cea1cbd947a733b563eac1b5/contracts/optimistic-ethereum/OVM/precompiles/OVM_L1MessageSender.sol#L12-L13

But I like that more granular standard approach you suggest.